### PR TITLE
Add landing pages

### LINKED ISSUE #327: Adding Landing pages to the Template Library
As discussed [in Slack](https://vertistudio.slack.com/archives/C02JC68P404/p1664191646111599?thread_ts=1663579666.320889&cid=C02JC68P404), we are going to add landing pages to the template library of the plugin.

Initially, we are going to introduce a generic `Landing page` category where we will include templates from different niches/subcategories. 

Below is a screenshot from the [Figma file](https://www.figma.com/file/r1pLMeeq4nhynBlCKY6mOQ/Coming-Soon-pages?node-id=509%3A1126) where I have been working on some mockups for these templates.

### Templates
Currently the collection will include:

- A personal / bio landing page
- A simple portfolio
- A Linktree-type page
- A wedding announcement page
- A service landing page
- An Event landing page

- One more Coming Soon page
- One more Maintenance page

I will be building the landing pages in the demo site, You should be able to find them [in this page.](https://demosites.io/otter/landing-pages/)

landing pages

### Monitoring user preferences through API
As Marius mentioned in Slack, 

> would be super great if we can send the page type selected, both the name and the category on the api when they subscribe with email. in this way we can monitor which category/page they like more.

### Changes in the Wizard

We need to update the Wizard flow, and include a Landing Page option.

### Review for potential changes in the Plugin Settings

We also need to review the current settings and see if we need to make any changes there. Some things we need to do are:

- [x] Include the new templates in the library (design tab)
- [x] Add the Landing page category in the dropdown (design tab)
- [x] Review main settings page and see if we need to make any meaningful changes here, in case the user has selected a Landing page. 

### Next Steps

From my side, I will start building the templates, and -in parallel- work on the Wizard tweaks in order for @arinaturcu to be able to work on it while I complete the design of all templates.

Keeping you in the loop @selul @HardeepAsrani @arinaturcu @mghenciu 

Please let me know your input, suggestions or questions.

### PR DESCRIPTION
**Pull request by arinaturcu**
Adds the new Landing Pages.

- The new templates are shown in the gallery
-  The `Landing Page` category is present in the category dropdown
- New Landing Page template in the wizard
- The selected page has a more specific page status (instead of `Maintenance Page` the page will have `Maintenance \ Coming soon \ Landing Page` depending on the category)

Closed #327.

### CODE REVIEW

File: includes/classes/wp-maintenance-mode-admin.php [hunk 1]
```diff
@@ -647,7 +647,7 @@ public function insert_template() {
 				$post_arr['ID'] = $this->plugin_settings['design']['page_id'];
 				$page_id        = wp_update_post( $post_arr );
 			} else {
-				$post_arr['post_title'] = 'Maintenance Page';
+				$post_arr['post_title'] = __( 'New Page', 'wp-maintenance-mode' );
```
Comment (selul) [inline includes/classes/wp-maintenance-mode-admin.php:650]: we can still use the `Maintenance Page` as default, translated. I think is more straightforward than New Page, even if the user selected a different category.

**Review by selul — CHANGES_REQUESTED** [review]

File: includes/classes/wp-maintenance-mode-admin.php [hunk 2]
```diff
@@ -695,9 +696,10 @@ public function subscribe_newsletter() {
 					),
 					'body'    => wp_json_encode(
 						array(
-							'slug'  => 'wp-maintenance-mode',
-							'site'  => get_site_url(),
-							'email' => $_POST['email'],
+							'slug'     => 'wp-maintenance-mode',
+							'site'     => get_site_url(),
+							'email'    => $_POST['email'],
+							'category' => get_option( 'wpmm_page_category' ),
```
Comment (selul) [inline includes/classes/wp-maintenance-mode-admin.php:702]: another small thing, can we add the category under data prop, i.e [ data => category => get_option ... ]

**Review by selul — CHANGES_REQUESTED** [review]

File: includes/classes/wp-maintenance-mode-admin.php (hunk 2 above)
Comment (arinaturcu) [reply includes/classes/wp-maintenance-mode-admin.php:702]: right, thank you!

**Review by selul — APPROVED** [review]

**Comment by JohnPixle** [comment]
I just tested the build on a TasteWP instance and it works perfect, nice Job @arinaturcu 

I only noticed that the compression of the library images is pretty harsh. Not a critical issue, but perhaps we can do something about it (lowering the compression)? Happy to help with the screenshots if you need me to, I am not sure if these are auto-generated 🤔  

Screenshot 2022-10-26 at 13 50 26

**Comment by arinaturcu** [comment]
Thank you for testing @JohnPixle!
They are not auto generated, I took them. If you think they are OK in terms of structure, I can lower the compression and replace them.
Otherwise, some new ones are welcomed 🚀

**Comment by arinaturcu** [comment]
@JohnPixle Okay, I'll leave them to you then. Thank you for helping with these 🚀

**Comment by JohnPixle** [comment]
Hey again @arinaturcu 

So, I took a deeper look with the thumbnails, I think the issue with the overly sharp edges was coming from the resizing of the images, and not the actual compression. The [original images](https://energeticglue.s3-tastewp.com/wp-content/plugins/wp-maintenance-mode/assets/templates/landing-page/portfolio/screenshot.png) appear to be big, and when resized they appear with the rough edges.

I took the initiative to provide the images to you in the final dimensions as they appear in the library, 380 x 405. I hope the dimensions are correct. If not, let me know.

I also safely compressed them a bit. Now all the thumbnails of the landing pages are only 174 KB, which is significantly less than before (an original image of a single landing page template used to be about 400KB). Hope it helps 👍🏻 

I did this for all 3 thumbnail types, Landing pages, Maintenance and coming soon. I am attaching the Zips here.

[landing-pages.zip](https://github.com/Codeinwp/wp-maintenance-mode/files/9877964/landing-pages.zip)
[Coming Soon.zip](https://github.com/Codeinwp/wp-maintenance-mode/files/9877975/Coming.Soon.zip)
[maintenance.zip](https://github.com/Codeinwp/wp-maintenance-mode/files/9877984/maintenance.zip)

I would also like to point out two things:

**1. New templates.** Apart from the landing pages, in this version of the plugin, we have one new Coming Soon template. [Link here.](https://demosites.io/otter/coming-soon-pages/coming-soon-4).  We also have one more Maintenance template. [Link here](https://demosites.io/otter/coming-soon-pages/maintenance-4). The thumbnails are included in the zips. 

**2. Template names.** For the Maintenance and Coming Soon templates I would advise to start using a bit more generic names (maintenance 1, maintenance 2 etc) instead of style-related names as we do now (modern , cover etc). I am suggesting this in order to keep things easy since we are going to add more of these templates in the future, and we may not be able to define a unique style name for each of them. 

Let me know what you think and always happy to help for anything you may need.

**Comment by arinaturcu** [comment]
Great, @JohnPixle! 🚀
I will replace the screenshots and the names of the templates and add the new ones. I will come back with an update afterward.

Thank you!

**Comment by irinelenache** [comment]
@arinaturcu Tested and didn't find any problem in terms of functionality 🚀 

After @JohnPixle's mentions are addressed and he approves it, this can be moved to **Ready to merge** ✅

**Comment by JohnPixle** [comment]
@arinaturcu Thanks for the update, we are all good with the Thumbnails quality, thanks so much for taking care of it 🙂 👍🏻

I noticed two small issues though:

### Wizard:
- [ ] When I run the wizard, the image links are broken. Maybe it has to do with the renaming of the files? Tried on a new tasteWP instance.

- [ ] Also, in the description, I would suggest that instead of **pre-written** templates we use the term **pre-designed** templates.

Screenshot 2022-10-28 at 10 54 53

### Ordering of the templates

- [ ] It's not critical, but I was wondering if we can have the maintenance / coming soon templates appear in the correct order (Maintenance 1, Maintenance 2, Maintenance 3 etc...). Landing pages are ok.


Screenshot 2022-10-28 at 10 47 33
Screenshot 2022-10-28 at 10 47 45

Let me know your thoughts and don't hesitate to ping me in case you need any clarifications on the above.

**Comment by arinaturcu** [comment]
@JohnPixle Thank you for catching these 😄 .
Should be fixed now.

**Comment by JohnPixle** [comment]
@arinaturcu Just checked and all looks rock-solid. Thanks so much for taking care of them, looking forward to see them live. From my side it's a green light to merge! 🚀

**Review by selul — APPROVED** [review]
🚀

**Pull request merged**